api/app/reviews/tests.py

```diff
@@ -1484,6 +1488,3 @@ def test_get_reference_request_with_two_references(self):
         self.assertDictEqual(data['references'][1], REFERENCE_DETAIL_2)
         self.assertEqual(data['references'][1]['token'], reference_req2.token)
         self.assertEqual(data['references'][1]['uploaded_document'], 'DOCT-UPLOAD-78979')
-

```

Comment:
@avishkar58 I am getting quite a few "foreign key constraint" errors with the test. The current one is for lines 1436 and 1463.
I'm just wondering if it's correct to continue checking the `static_seed_data`? I'm not sure when to move away from fixing the test :)

Reply:
Ah so the reason for that is because the first constructor parameter of ReferenceRequest is a response_id and the value of 1 that you're passing here doesn't correspond to a Response because of line 1420 which is commented out :) So the test_response that you create inside seed_static_data isn't actually being added to the database. In fact, instead of passing 1 in the ReferenceRequest constructor, it would be better to pass `self.test_response.id`  -  because there's actually no guarantee that the first response you create will have id of 1, using `self.test_response.id` will guarantee that you're referring to the correct response object 

`reference_req = ReferenceRequest(self.test_response.id, 'Mr', 'John', 'Snow', 'Supervisor', 'common@email.com')`

Regarding when to move away from fixing the test, that should only be when the test is failing on one of your assertion lines and for a reason that you expect it to (ie that the feature you're testing hasn't been implemented yet). If the test is failing in the setup code then you definitely need to keep fixing the test :)